src/main/java/com/ecommerce/ECommerceApp/model/Product.java

```diff
@@ -1,70 +1,134 @@
 package com.ecommerce.ECommerceApp.model;
 
+import org.springframework.stereotype.Component;
+
+import java.nio.file.Path;
+import java.util.List;
+
+/**
+ * Represents a product in the ECommerce system.
+ * Contains: productID, productName, productDescription, ProductPrice, productStock, productCategory, productSubcategory
+ */
 @Component
 public class Product {
     private int productID;
-    private string productName;
-    private string productDescription;
+    private String productName;
+    private String productDescription;
     private float productPrice;
     private int productStock;
-    private choice productCategory;
-    private choice productSubCategory;
-    private array productImage;
-
-
+    private Category productCategory;
+    private Category productSubCategory;
+    private List<Path> productImage;
```

Comment:
Changed from an array into a list containing path. This will enable image paths to be used to display multiple images, and since it is a list instead of an array, the contents can be more dynamic than an already initialized array. This will help with varied amounts of images per product. 

Source: https://www.capitalone.com/tech/software-engineering/10-microservices-best-practices/